packages/infer-schema/src/utils/unique-string.ts

```diff
@@ -0,0 +1,28 @@
+/*
+ * Copyright (c) "Neo4j"
+ * Neo4j Sweden AB [http://neo4j.com]
+ *
+ * This file is part of Neo4j.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+export default function uniqueString(candidate: string, pool: string[]): string {
+    let uniqueStr = candidate;
+    let counter = 2;
+    while (pool.includes(uniqueStr)) {
+        uniqueStr = candidate + String(counter);
+        counter += 1;
+    }
+    return uniqueStr;
+}
```

Comment:
Can a comment be added as to the intention of this function? Maybe a more descriptive function name as well?

Reply:
👍